presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -461,9 +461,9 @@ private synchronized void setQueryOutputInfo(QueryExecution.QueryOutputInfo outp
         return Futures.transformAsync(queryManager.getStateChange(queryId, currentState), this::queryDoneFuture);
     }
 
-    private synchronized URI createNextResultsUri(UriInfo uriInfo)
+    private synchronized URI createNextResultsUri(String scheme, UriInfo uriInfo)
     {
-        return uriInfo.getBaseUriBuilder().replacePath("/v1/statement").path(queryId.toString()).path(String.valueOf(resultId.incrementAndGet())).replaceQuery("").build();
+        return uriInfo.getBaseUriBuilder().replacePath("/v1/statement").path(queryId.toString()).path(String.valueOf(resultId.incrementAndGet())).replaceQuery("").scheme(scheme).build();
```

Comment:
Replace scheme first since it's first in the URL. Let's also wrap this since it's getting long:
```java
return uriInfo.getBaseUriBuilder()
        .scheme(scheme)
        .replacePath("/v1/statement")
        .path(queryId.toString())
        .path(String.valueOf(resultId.incrementAndGet()))
        .replaceQuery("")
        .build();

Reply:
There is another spot that builds the URL for `query.html`. This is returned as part of the query stats and is printed by the CLI when run with `--debug` mode. Do we need to change that as well?

Reply:
I fixed this for `query.html` as well.